### PR TITLE
Add revisions

### PR DESCRIPTION
**Pull request by richdynamix**
This PR adds the revisions to the driver.

- [x] Working Copy is now latest revision

### CODE REVIEW

**Review by rodolfo-perez — APPROVED** [review]

File: src/Http/Controllers/CP/Collections/RestoreEntryRevisionController.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+    public function __invoke(Request $request, $collection, $entry)
+    {
+        $date = Carbon::parse($request->revision);
+        $revisionModel = RevisionModel::where('date', $date)->first();
+
+        if (! $target = $entry->revision($revisionModel->id)) {
+            dd('no such revision', $request->revision);
+            // todo: handle invalid revision reference
+        }
```
Comment (falko189) [inline src/Http/Controllers/CP/Collections/RestoreEntryRevisionController.php:23]: DD here?

File: src/Http/Controllers/CP/Collections/RestoreEntryRevisionController.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        // Update the entry with the targeted revision data.
+        if ($entry->published()) {
+            $workingCopy = WorkingCopy::fromModel($revisionModel)->date(now());
+            $workingCopy->save();
+
+            $this->updateWorkingCopyRevision($workingCopy, $entry);
+        } else {
+            $entry->makeFromRevision($target)->published(false)->save();
+            $this->updateWorkingCopyRevision($target, $entry);
+        }
```
Comment (falko189) [inline src/Http/Controllers/CP/Collections/RestoreEntryRevisionController.php:34]: Could we do if {
//code
return
}

File: src/Http/Controllers/CP/Collections/RestoreEntryRevisionController.php (hunk 1 above)
Comment (richdynamix) [reply src/Http/Controllers/CP/Collections/RestoreEntryRevisionController.php:23]: Was copied from the cms controller with method override. I guess we can ignore it here.

File: src/Http/Controllers/CP/Collections/RestoreEntryRevisionController.php (hunk 2 above)
Comment (richdynamix) [reply src/Http/Controllers/CP/Collections/RestoreEntryRevisionController.php:34]: Potentially yes, again it was copied from the original. Would need to duplicate the flash data call though.

**Pull request merged**